modules/integration/test-common/integration-test-utils/src/main/java/org/wso2/esb/integration/common/utils/MailToTransportUtil.java

```diff
@@ -0,0 +1,335 @@
+import javax.mail.PasswordAuthentication;
+import javax.mail.Session;
+import javax.mail.Store;
+import javax.mail.search.AndTerm;
+import javax.mail.search.BodyTerm;
+import javax.mail.search.FlagTerm;
+import javax.mail.search.SearchTerm;
+import javax.mail.search.SubjectTerm;
+import javax.xml.xpath.XPathExpressionException;
+import java.rmi.RemoteException;
+import java.util.Properties;
+
+/**
+ * This Provides utility methods to sending, receiving and verifying emails by reading logs
+ * By using javax mail api to send emails and LogViewerClient to read the log
+ */
+public class MailToTransportUtil {
+
+    protected static Log log = LogFactory.getLog(MailToTransportUtil.class);
+    private static String sender;
+    private static char[] senderPassword;
+    private static String receiver;
+    private static char[] receiverPassword;
+    private static String domain;
+    private static int WAIT_TIME_MS = 180 * 1000; // Max time to wait for a email and string search in log
+    private static final String INBOX = "INBOX";
+    private static final String EMAIL_CREDENTIAL_PARENT_XPATH = "//emailCredentials";
+    private static final String EMAIL_CREDENTIAL_SENDER_XPATH = "//emailCredentials/sender";
+    private static final String EMAIL_CREDENTIAL_SENDER_PASSWORD_XPATH = "//emailCredentials/senderPassword";
+    private static final String EMAIL_CREDENTIAL_RECEIVER_XPATH = "//emailCredentials/receiver";
+    private static final String EMAIL_CREDENTIAL_RECEIVER_PASSWORD_XPATH = "//emailCredentials/receiverPassword";
+    private static final String EMAIL_CREDENTIAL_DOMAIN_XPATH = "//emailCredentials/domain";
+
+    /**
+     * @return SMTP properties of Gmail server
+     */
+    public static Properties getSMTPProperties() {
+        Properties props = new Properties();
+        props.put("mail.smtp.host", "smtp.gmail.com");
+        props.put("mail.smtp.auth", "true");
+        props.put("mail.smtp.debug", "true");
+        props.put("mail.smtp.socketFactory.port", "465");
+        props.put("mail.smtp.socketFactory.class", "javax.net.ssl.SSLSocketFactory");
+        props.put("mail.smtp.socketFactory.fallback", "false");
+        props.put("mail.smtp.starttls.enable", "true");
+        return props;
+    }
+
+    /**
+     * Send email and check email received successfully
+     *
+     * @param emailSubject - Massage to send in email
+     * @return Email - Email sent successfully or not
+     * @throws Exception - Error while sending the gmail
+     */
+    public static boolean sendMailAndCheckReceived(String emailSubject) throws Exception {
+        Properties props = getSMTPProperties();
+        EmailSender emailSender =
+                new EmailSender(props, sender, String.valueOf(senderPassword), domain, receiver + "@" + domain);
+        if (emailSender.createSession()) {
+            emailSender.setSubject(emailSubject);
+            emailSender.setBody("Body : " + emailSubject);
+            emailSender.sendEmail();
```

Comment:
sendEmail throws Exception and we need to fix it in next framework release to send custom or specific exception properly.